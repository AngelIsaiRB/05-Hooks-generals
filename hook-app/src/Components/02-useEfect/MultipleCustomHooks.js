import React from 'react'
import { UseCouter } from '../../hooks/UseCouter'
import { useFetch } from '../../hooks/useFetch'
import "../02-useEfect/effects.css"

export const MultipleCustomHooks = () => {
    const {state,increment, decrement,reset,}=UseCouter(1);
   const {loading, data} = useFetch(`https://www.breakingbadapi.com/api/quotes/${state}`);
   const {author, quote} = !!data && data[0];

    
    return (
        <div>
            <h1>Braking bad quotes </h1>
            <hr/>
            {
                loading
                ?
                    (
                        <div className="alert alert-info text-center">
                        loading...
                        </div>
                    )
                :
                    (
                        <blockquote className="blockquote text-right">
                        <p className="mb-0">{quote} </p>
                        <footer className="blockquote-footer">{author} </footer>
                        </blockquote>
                    )
            }
            <button className="btn btn-primary" onClick={()=>increment(1)}>
                next quote
            </button>
            
            
        </div>
    )
}
