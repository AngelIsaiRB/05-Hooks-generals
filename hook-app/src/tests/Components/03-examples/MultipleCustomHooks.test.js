import { shallow } from 'enzyme'
import React from 'react'
import { MultipleCustomHooks } from '../../../Components/03-examples/MultipleCustomHooks';
import { UseCouter } from '../../../hooks/UseCouter';
import { useFetch } from '../../../hooks/useFetch';

jest.mock('../../../hooks/useFetch');
jest.mock('../../../hooks/UseCouter');


describe('test en multiple custom hooks', () => {
  

   beforeEach(()=>{
    UseCouter.mockReturnValue({
        state: 10,
        increment: () => {}
    });
   })
    


    test('should de mostrarse correctamente', () => {

 

            useFetch.mockReturnValue({
                data:null,
                loading:true,
                error:null
            });


        const wrapper = shallow(<MultipleCustomHooks/>);
        expect(wrapper).toMatchSnapshot();
    });

    test('should de mostrar la informacion ', () => {    
        useFetch.mockReturnValue({
            data:[{
                    author:'isai',
                    quote:'hola mundo'
                }],
                loading: false,
                error:null
            });                        
            const wrapper = shallow( <MultipleCustomHooks /> );    
        expect( wrapper.find('.alert').exists() ).toBe( false );
        expect(wrapper.find(".mb-0").text().trim()).toBe("hola mundo");
        expect(wrapper.find("footer").text().trim()).toBe("isai");

    })
    
    
    
})
