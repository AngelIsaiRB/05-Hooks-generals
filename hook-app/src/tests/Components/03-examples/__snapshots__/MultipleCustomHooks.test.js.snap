// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test en multiple custom hooks should de mostrarse correctamente 1`] = `
<div>
  <h1>
    Braking bad quotes 
  </h1>
  <hr />
  <div
    className="alert alert-info text-center"
  >
    loading...
  </div>
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    next quote
  </button>
</div>
`;
